Extensions/ShareTo/ShareViewController.swift

```diff
@@ -84,16 +84,34 @@ class ShareViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        view.backgroundColor = .white
-
+        checkTheme()
+        setupUI()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        checkTheme()
+        setupUI()
+    }
+    
+    func checkTheme() {
+        if #available(iOS 13, *) {
+            ThemeManager.instance.current = traitCollection.userInterfaceStyle == .dark ? DarkTheme() : NormalTheme()
+        } else {
+            ThemeManager.instance.current = NormalTheme()
+        }
+    }
```

Comment:
I will recommend you a better approach using `UIColor.init(dynamicProvider:)` that's avoid implementing `traitCollectionDidChange` and `checkTheme`.

```swift

public struct ModernColor {
    var darkColor: UIColor
    var lightColor: UIColor
    
    public init(darkColor: UIColor, lightColor: UIColor) {
        self.darkColor = darkColor
        self.lightColor = lightColor
    }
    
    public var color: UIColor {
        if #available(iOS 13, *) {
            return UIColor { (traitCollection: UITraitCollection) -> UIColor in
                if traitCollection.userInterfaceStyle == .dark {
                    return self.color(for: .dark)
                } else {
                    return self.color(for: .light)
                }
            }
        } else {
            return color(for: .light)
        }
    }
    
    private func color(for scheme: ColorScheme) -> UIColor {
        return scheme == .dark ? darkColor : lightColor
    }
}

public protocol Theme { // or directly your `struct`
    var doneLabelBackground: ModernColor { get }
    var separator: ModernColor { get }
    var actionRowTextAndIcon: ModernColor { get }
    var defaultBackground: ModernColor { get }
}

```

Add your implementation, then just use `UIColor.theme.defaultBackground.color` and iOS will do the job for you.
IMHO it's easy to use and well readable.

Reply:
Yes absolutely you r right. Well we could also use colorAsset in the shareToExtension. What do you think?